src/reynir_correct/errtokenizer.py

```diff
@@ -2046,20 +2102,23 @@ def is_valid_suggestion(
         # !!! TODO: Handle upper/lowercase
         if token.txt in UniqueErrors.DICT:
             # Note: corrected is a tuple
+            rtok = token
             corrected = UniqueErrors.DICT[token.txt]
             assert isinstance(corrected, tuple)
             corrected_display = " ".join(corrected)
             for ix, corrected_word in enumerate(corrected):
                 if ix == 0:
-                    rtok = replace_word(1, token, corrected_word, corrected_display)
+                    if "S001" not in ignore_rules:
```

Comment:
This code looks quite suspicious. I suggest that if S001 is to be ignored, this loop is not executed at all; i.e. that the if statement to check for ignoring S001 is in an outer scope and that there is an earlier exit from the function in that case.

Reply:
This was an attempt to better isolate the switch to make its behavior clearer. Check is now outside the loop, which is not executed if the check fails.